include/cv/dynamic_detection.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+// #ifndef INCLUDE_CV_DYNAMIC_DETECTION_HPP_
+// #define INCLUDE_CV_DYNAMIC_DETECTION_HPP_
+
+#include <mutex>
+#include <queue>
+#include <chrono>
+#include "utilities/constants.hpp"
+#include "utilities/locks.hpp"
+#include <thread>
+#include "camera/interface.hpp"
+
+using namespace std::chrono_literals;
+
+/*
+Design:
+Producer-consumer thread communication model so that dynamic detection can run
+asyncronously on its own to detect the forward camera for obstacles ~1Hz.
+
+Steps:
+1. Front camera takes pictures every second, puts into workqueue*.
+2. FastSAM runs on the queue.
+3. dynamic_detection looks at the segmentation output and decides
+   whether to enter emergency avoidance.
+4. Tick control gets handed over to the dynamic avoidance state.
+
+(*) Not totally sure if I will end up putting a queue here. Note that
+    if Thread 2 is using the Detection queue, it can lock out Thread 1's
+    execution.
+
+Threads:
+Thread 1: Runs steps 1-3.
+Thread 2: This is the main program execution, which should be polling every tick.
+*/
+
+// Output after an obstacle has been detected
+struct Detection {
+    std::chrono::time_point<std::chrono::system_clock> time;
+    // TODO(Samir): is there a type for location of bounding box
```

Comment:
There is a bounding box but it stores two corners of the bounding box, not the center of the bounding box.

https://github.com/tritonuas/obcpp/blob/664bbc81eb86ff18b0b7a647ade1d08fb56769cf/include/cv/utilities.hpp#L20-L29